code/modules/hydroponics/compost_bin.dm

```diff
@@ -0,0 +1,195 @@
+#define BASE_BIOMASS_CAPACITY 1500
+#define BASE_COMPOST_CAPACITY 1500
+#define SOIL_COST 25
+#define SOIL /obj/item/stack/sheet/soil
+#define DECAY 0.2
+#define MIN_CONVERSION 10
+/**
+  * # compost bin
+  * used to make soil from plants.
+  * Doesn't have components.
+  */
+/obj/machinery/compost_bin
+	name = "compost bin"
+	desc = "A wooden bin for composting."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "compost_bin-empty"
+	power_state = NO_POWER_USE
+	density = TRUE
+	anchored = TRUE
+	/// amount of biomass in the compost bin
+	var/biomass = 0
+	/// amount of compost in the compost bin
+	var/compost = 0
+	/// The maximum amount of biomass the compost bin can store.
+	var/biomass_capacity = BASE_BIOMASS_CAPACITY
+	/// The maximum amount of compost the compost bin can store.
+	var/compost_capacity = BASE_COMPOST_CAPACITY
+	var/composting = FALSE
+
+/obj/machinery/compost_bin/Initialize(mapload)
+	/// try to compost
+	compost()
+	return ..()
+
+
+/obj/machinery/compost_bin/on_deconstruction()
+	/// returns wood instead of the non-existent components
+	new /obj/item/stack/sheet/wood(loc, 10)
+	..()
+
+/obj/machinery/compost_bin/screwdriver_act(mob/living/user, obj/item/I)
+	/// there are no screws either
+	to_chat(user, "<span class='warning'>[src] has no screws!</span>")
```

Comment:
Return true here to prevent them from hitting it with a screwdriver
```suggestion
	to_chat(user, "<span class='warning'>[src] has no screws!</span>")
	return TRUE
```